## DAY 2 BOOT CAMP — SESSION 1 ##
Python Input + Data Types<br>
Today you learn how programs receive information, store different kinds of values, and calculate results.

Topic 1: User Input

A program can ask the user for data. This is called user input. In Python, we can use the `input()` function to get input from the user. The `input()` function takes a string as an argument, which is the prompt that will be displayed to the user. The function then waits for the user to enter some data and press Enter. The data entered by the user is returned as a string.

In [1]:
name = input("Enter material name: ")
print(name)

Steel


## Important Rule ##

input() always gives text (string).<br>
Even if user types: 25 , python stores "25". That matters for calculations

## Topic 2: Numbers from Input ##

To calculate, convert text into numbers.

In [2]:
temp = float(input("Enter temperature: "))
print(temp)

786.54


Why float()?<br>
Use float() when decimals are possible.<br>
Examples:<br>
density = 2.7<br>
voltage = 12.5<br>
pressure = 101.3<br>
int() for Whole Numbers

In [3]:
atoms = int(input("Enter atoms count: "))
print(atoms)

50


## Topic 3: Data Types

Everything in Python has a type<br>
| Type    | Meaning        | Example    |
| ------- | -------------- | ---------- |
| Integer | Whole number   | `5`        |
| Float   | Decimal number | `3.14`     |
| String  | Text           | `"Copper"` |
| Boolean | True/False     | `True`     |


In [4]:
#Check Type
x = 5
print(type(x))

<class 'int'>


In [5]:
a = 2.7
print(type(a))

<class 'float'>


In [6]:
name = "Steel"
print(type(name))

<class 'str'>


In [7]:
safe = True
print(type(safe))

<class 'bool'>


## Topic 4: Arithmetic in Python

In [8]:
a = 10
b = 3

print(a+b)
print(a-b)
print(a*b)
print(a/b)
print(a**2)

13
7
30
3.3333333333333335
100


## Results

| Expression | Meaning  | Result |
| ---------- | -------- | ------ |
| a+b        | Add      | 13     |
| a-b        | Subtract | 7      |
| a*b        | Multiply | 30     |
| a/b        | Divide   | 3.3333 |
| a**2       | Power    | 100    |


In [9]:
# Scientific Example
radius = 5
area = 3.14 * radius**2
print(area)

78.5


Topic 5: Comments<br>
Comments are notes for humans.
Python ignores comments.
Use them to explain formulas, units, assumptions.


In [10]:
# density of aluminum (comment)
density = 2.7

In [11]:
# Good Scientific Style

# density in g/cm^3
density = 2.7

# convert cm to m
length_m = 25 / 100

## EXERCISES (With Solutions)

In [12]:
# Exercise 1
# Ask user for material name and print it.

material = input("Enter material name: ")
print(material)

aluminium


In [13]:
# Exercise 2
# Ask for density and print:
# Density entered = ...
density = float(input("Enter density: "))
print("Density entered =", density)

Density entered = 7.8


In [14]:
# Exercise 3
# Compute square of 7.
print(7**2)

49


## Mini Scientific Practice

In [15]:
material = input("Material: ")
density = float(input("Density: "))

print("Material =", material)
print("Density =", density, "g/cm^3")

Material = Steel
Density = 7.8 g/cm^3
